### ISSUE TITLE
add Iterable<Listenable> .watch extension

### ISSUE THREAD

**Issue by lukepighetti** [s0nerik/context_plus#15]
<img width="649" alt="Screenshot 2024-08-09 at 3 43 21 PM" src="https://github.com/user-attachments/assets/2ba8a68a-0ecc-4f4f-98b1-411a811f6694">

```dart
extension on Iterable<Listenable> {
  void watch(BuildContext context) => Listenable.merge(this).watch(context);
}
```

**Comment by s0nerik** [comment]
I like the general idea you're proposing and I see how it can potentially improve code readability by removing the repetitive .watch() calls.

The proposed implementation, though, not as much.

Using `Listenable.merge()` there will result in a new subscription creation on each build and disposal before the next one. That's because an instance of the listenable changes each build.

Another problem with the proposed implementation is that it's impossible to retrieve the current state of such mass-watched listenables from the method result.

I believe it would be better to have such API defined on a [record](https://dart.dev/language/records#record-syntax) instead so that the specific types of each listenable in the list are known and can be used for determining the value to be returned. Something similar to how `(future1, future2).wait` is defined.

I'll look into adding this after I'm done with introducing side effects into context_plus.

Thanks for a good idea!

**Comment by s0nerik** [comment]
The day has come! Mass-watching is now available in [context_watch 5.1.0](https://pub.dev/packages/context_watch/changelog#510) and [context_plus 4.1.0](https://pub.dev/packages/context_plus/changelog#410)!

You can combine up to 4 observables (or observable Refs) into a record and call `.watch()`, `.watchOnly()` or `.watchEffect()` on them.
